### DISCUSSION TITLE
Assign further file extension to a language support

### DISCUSSION THREAD

**Discussion by white-gecko** [pulsar-edit/.github#267] [Pulsar Support]
Dear all,
I currently create a presentation using [quarto](https://quarto.org/) which uses a syntax derived from markdown with some additions. They use the `.qmd` file extension. I don't expect pulsar to support every kind of new syntax/language/file formats but still having a Markdown syntax highlighting would really help in working with `.qmd` files. Is there any way to tell markdown to apply some syntax highlighting to some additional set of file extensions?
In vim I would just do `set filetype=markdown` or add it to a modeline. (actually vim already has a filetype for quarto but don't mind).

Is there any such feature?

Thank you very much!

**Comment by confused-Techie** [comment]
There's actually a few options that might be relevant here:

If you want to change the grammar used or a specific file extension **one** time, the Grammar Selector is a great choice. By default in the lower right of the status bar you'll see the grammar selector where you can click to change the currently active grammar.
![image](https://github.com/pulsar-edit/.github/assets/26921489/2a384b83-1f08-4588-90ba-3a33db441efe)

Notice the highlighted section in the photo. You can click this to change your currently active grammar to any language you want.

Otherwise if you want to change the grammar you are using **always** then you can follow along with the [docs](https://pulsar-edit.dev/docs/launch-manual/sections/using-pulsar/#customizing-language-recognition) to do this. Just a simple entry into your config file.

The last thing that you might want to know in this case, since we are talking about `markdown`, if you are using the `markdown-preview` package and want to use it for this new extension make sure to update the list of `Grammars` in the `markdown-preview` package's settings.

**Reply by white-gecko** [reply]
Thank you for pointing me to these places!

**Reply by white-gecko** [reply]
In the [docs](https://pulsar-edit.dev/docs/launch-manual/sections/using-pulsar/#customizing-language-recognition) it says:

> In the example above, source.coffee is the language's scope name (see [Finding a Language's Scope Name](https://pulsar-edit.dev/docs/launch-manual/sections/using-pulsar/#finding-a-language-s-scope-name) for more information) and foo is the file extension to match without the period. Adding a period to the beginning of either of these will not work.

And then under [Finding a Language's Scope Name](https://pulsar-edit.dev/docs/launch-manual/sections/using-pulsar/#finding-a-language-s-scope-name):

> The scope name is shown in the settings view for each language. Click on "Packages" in the navigation on the left, search for the language of your choice, select it, and you should see the scope name under the language name heading

But I can't find any langue package for markdown, it appears to be right in the core of pulsar. Is that right?

Anyhow, `source.markdown` seems to be the right identifier.

**Reply by confused-Techie** [reply]
@white-gecko You should still be able to find the Markdown package, it is within the core of Pulsar, but the core of Pulsar is just a bunch of third party packages.

The default markdown package is titled `language-gfm` for GitHub Flavored Markdown, probably why you had some trouble finding it. Anyways if you've enabled Pulsar's URI just click the following link and it'll take you to the package in settings: [`language-gfm` settings]  (atom://config/packages/language-gfm). (Seems the link was displaying properly after posting, so just adding a space here so you can manually copy and paste this link into any web browsers address bar)
